lib/eth/abi/event.rb

```diff
@@ -24,20 +24,69 @@ module Abi
     module Event
       extend self
 
+      # Compute topic for ABI event interface.
+      #
+      # @param interface [Hash] ABI event interface.
+      # @return [String] a hex-string topic.
+      def compute_topic(interface)
+        sig = Abi.signature(interface)
+        Util.prefix_hex(Util.bin_to_hex(Util.keccak256(sig)))
+      end
+
+      # Decodes a stream of receipt logs with a set of ABI interfaces.
+      #
+      # @param interfaces [Array] event ABI types.
+      # @param logs [Array] transaction receipt logs
+      # @return [Hash] an enumerator of decoded log objects.
+      def decode_logs(interfaces, logs)
+        Enumerator.new do |y|
+          topic_to_interfaces = Hash[interfaces.map { |i| [compute_topic(i), i] }]
+
+          logs.each do |log|
+            result = { log: log }
```

Comment:
I wonder if this *decoded log hash* should be it's own proper class or struct? ethers.js calls theirs [`LogDescription`](https://docs.ethers.io/v5/api/utils/abi/interface/#LogDescription).

Reply:
Added the log description struct to 07b0dcfc78c31ffa46bcced39a82cf96ae19208f